Samples/BatchSupport/BatchSupport/SampleHelper.cpp

```diff
@@ -0,0 +1,221 @@
+  wchar_t drive[_MAX_DRIVE];
+  wchar_t dir[_MAX_DIR];
+  wchar_t filename[_MAX_FNAME];
+  wchar_t ext[_MAX_EXT];
+  _wsplitpath_s(modulePath, drive, _MAX_DRIVE, dir, _MAX_DIR, filename,
+                _MAX_FNAME, ext, _MAX_EXT);
+
+  val = drive;
+  val += dir;
+
+  return val;
+}
+
+std::vector<float>
+SoftwareBitmapToSoftwareTensor(SoftwareBitmap softwareBitmap) {
+  /* Manully tensorize from CPU resource, steps:
+  1. Get the access to buffer of softwarebitmap
+  2. Transform the data in buffer to a vector of float
+  */
+
+  // 1. Get the access to buffer of softwarebitmap
+  BYTE *pData = nullptr;
+  UINT32 size = 0;
+  BitmapBuffer spBitmapBuffer(
+      softwareBitmap.LockBuffer(BitmapBufferAccessMode::Read));
+  winrt::Windows::Foundation::IMemoryBufferReference reference =
+      spBitmapBuffer.CreateReference();
+  auto spByteAccess =
+      reference.as<::Windows::Foundation::IMemoryBufferByteAccess>();
+  CHECK_HRESULT(spByteAccess->GetBuffer(&pData, &size));
+
+  uint32_t height = softwareBitmap.PixelHeight();
+  uint32_t width = softwareBitmap.PixelWidth();
+  BitmapPixelFormat pixelFormat = softwareBitmap.BitmapPixelFormat();
+  uint32_t channels = BitmapPixelFormat::Gray8 == pixelFormat ? 1 : 3;
+
+  std::vector<int64_t> shape = {1, channels, height, width};
+  std::vector<float> outputVector(channels * height * width);
+
+  // The channels of image stored in buffer is in order of BGRA-BGRA-BGRA-BGRA.
+  // Then we transform it to the order of
+  // BBBBB....GGGGG....RRRR....AAAA(dropped)
+
+  // 2. Transform the data in buffer to a vector of float
+  if (BitmapPixelFormat::Bgra8 == pixelFormat) {
+    for (UINT32 i = 0; i < size; i += 4) {
+      // suppose the model expects BGR image.
+      // index 0 is B, 1 is G, 2 is R, 3 is alpha(dropped).
+      UINT32 pixelInd = i / 4;
+      outputVector[pixelInd] = (float)pData[i];
+      outputVector[(height * width) + pixelInd] = (float)pData[i + 1];
+      outputVector[(height * width * 2) + pixelInd] = (float)pData[i + 2];
+    }
+  } else if (BitmapPixelFormat::Rgba8 == pixelFormat) {
+    for (UINT32 i = 0; i < size; i += 4) {
+      // suppose the model expects BGR image.
+      // index 0 is B, 1 is G, 2 is R, 3 is alpha(dropped).
+      UINT32 pixelInd = i / 4;
+      outputVector[pixelInd] = (float)pData[i + 2];
+      outputVector[(height * width) + pixelInd] = (float)pData[i + 1];
+      outputVector[(height * width * 2) + pixelInd] = (float)pData[i];
+    }
+  } else if (BitmapPixelFormat::Gray8 == pixelFormat) {
+    for (UINT32 i = 0; i < size; i += 4) {
+      // suppose the model expects BGR image.
+      // index 0 is B, 1 is G, 2 is R, 3 is alpha(dropped).
+      UINT32 pixelInd = i / 4;
+      float red = (float)pData[i + 2];
+      float green = (float)pData[i + 1];
+      float blue = (float)pData[i];
+      float gray = 0.2126f * red + 0.7152f * green + 0.0722f * blue;
+      outputVector[pixelInd] = gray;
+    }
```

Comment:
can you remove the two cases that aren't used to keep the sample clean? this sample is more about batching than about tensorization, so we can just use the format the the images are already in.